File: 프로그래머스/Python/1단계/같은 숫자는 싫어.py
# 배열 arr에서 연속적으로 나타나는 숫자는 제거하고 남은 수들을 return 하는 solution 함수를 완성해 주세요.
# arr = [1, 1, 3, 3, 0, 1, 1] 이면 [1, 3, 0, 1] 을 return 합니다.


# 내 풀이
def solution(arr):
    answer =[arr[0]]
    for i in range(1, len(arr)):
        if arr[i] != arr[i-1]:
            answer.append(arr[i])
    return answer


# 다른사람 풀이
def no_continuous(s):
    a = []
    for i in s:
        if a[-1:] == [i]: continue  # [-1]을 통해 비교하면 empty array일 때 인덱싱오류가나서 [-1:]를 통해 리스트를 만들어 비교
        a.append(i)
    return a

File: 프로그래머스/JAVA/1단계/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> num = new ArrayList<>();
        num.add(arr[0]);
        for(int i = 1; i < arr.length; i++){
            if(num.get(num.size()-1) != arr[i]){
                num.add(arr[i]);
            }
        }
        int[] answer = new int[num.size()];
        for(int i = 0; i < num.size(); i++){
            answer[i] = num.get(i);
        }
        return num.toArray(new int[num.size()]);
    }
}
